Refonte blog : page d'accueil, images et liste des articles

AntoineAugusti opened the pull request:
Afficher les 3 derniers articles de blog sur la page d'accueil, tout en bas, sous forme de cartes.

![](https://puu.sh/DFOQA/2b521a7b27.png)

Live : https://deploy-preview-207--entrepreneur-interet-general.netlify.com/

Peut-être lié à #205

AntoineAugusti commented:
Test aussi avec des cartes sur le blog : https://deploy-preview-207--entrepreneur-interet-general.netlify.com/blog

guenolec commented:
J'ai fait des ajouts manuels de photos pour les 8 derniers articles de blogs, histoire de se faire une idée un peu plus réaliste de ce que cela donnerait. 

Preview ici : https://deploy-preview-207--entrepreneur-interet-general.netlify.com/blog/

AntoineAugusti commented:
Pour les images, la taille recommandée serait de 400x1040

AntoineAugusti commented:
Code to fill the images:

```py
import csv
import frontmatter

import glob

BASE_PATH = "/Users/antoineaugusti/Documents/site-eig/"

data = {}
with open("blog.csv") as f:
    reader = csv.DictReader(f)
    for row in reader:
        if row["photo"].startswith("/img") or row["photo"].startswith("http"):
            data[row["url"].split("/")[-1]] = row["photo"]

for file in glob.glob(BASE_PATH + "_posts/*.md"):
    current_image = None
    for post, image in data.items():
        if post.replace(".html", "") in file:
            current_image = image
    if current_image is None:
        continue
    with open(file) as f:
        content = frontmatter.load(f)
    content["image"] = current_image
    with open(file, "w") as f:
        frontmatter.dump(content, f)
```

AntoineAugusti commented:
To set tags

```py
import csv
import frontmatter

import glob

BASE_PATH = "/Users/antoineaugusti/Documents/site-eig/"

data = {}
with open("tags.csv") as f:
    reader = csv.DictReader(f)
    for row in reader:
        if not row["tags"].startswith("*"):
            data[row["url"].split("/")[-1]] = [
                t.strip() for t in row["tags"].split(",")
            ]

for file in glob.glob(BASE_PATH + "_posts/*.md"):
    current_tags = None
    for post, tags in data.items():
        if post.replace(".html", "") in file:
            current_tags = tags
    if current_tags is None:
        continue
    with open(file) as f:
        content = frontmatter.load(f)
    content["tags"] = current_tags
    with open(file, "w") as f:
        frontmatter.dump(content, f)
```
